### ISSUE TITLE
Performance doubts of a noob

### ISSUE THREAD

**Issue by RocketMike98** [JiehongLin/SAM-6D#14]
Really good job! Just managed to run it locally. I am really a noob in this field and my final goal is to have a code that allows me to infer real time the pose of an object. Looking at the performance of this type of algorithms, it seems like they take a lot of time to compute. In other words, if I'd like to run the provided code and see how it performs on a video or webcam, I would have to run this steps: 

python run_inference_custom.py --output_dir $OUTPUT_DIR --cad_path $CAD_PATH --rgb_path $RGB_PATH --depth_path $DEPTH_PATH --cam_path $CAMERA_PATH
export SEG_PATH=$OUTPUT_DIR/sam6d_results/detection_ism.json
cd ../Pose_Estimation_Model
python run_inference_custom.py --output_dir $OUTPUT_DIR --cad_path $CAD_PATH --rgb_path $RGB_PATH --depth_path $DEPTH_PATH --cam_path $CAMERA_PATH --seg_path $SEG_PATH

for every frame, right? Right now it takes a few seconds every time I do this two lines, so I guess that by running real time is not that realistic. Any feedback would be great!

**Comment by foollh** [comment]
The demo so far is just a simple stitching together of Instance Segmentation Model (ISM) and Pose Estimation Model(PEM). And use a large model to get better estimation accuracy. If you want to run through it faster, you can try the following points to improve it:
1. Changing SAM in ISM to Fast-SAM
2. Adjusting the hyperparameters of ISM to limit the number of proposals.
3. There are some overlapping functions in the two commands, such as feature extraction for templates, which can be saved before inference.
We may update the demo later or you can rewrite the code to speed up the inference by the above points. But we can't guarantee that the modification can achieve the real-time effect you want. You can see the inference time on BOP datasets in our paper, which can be a reference for you.

**Comment by RocketMike98** [comment]
Thanks for the feedback!! I'll give it a try :)

**Comment by foollh** [comment]
By large model I mean we use SAM-ViT-H as the backbone in ISM. In PEM, we only use ViT-B as the backbone. If you want to get faster speed, you can try the three points like what I mentioned above. If you don't want to reduce the performance of  ISM because it influences the PEM, you can try the third point mentioned above to get a speed boost.

> @foollh
> 
> The paper "SAM-6D" mentions the use of models trained on extensive datasets. Additionally, modifying the code related to the ISM might pose challenges, as the final value of the ISM influences the PEM. Could you suggest any methods to enhance the speed?
